### ISSUE TITLE
Average damage not correct for items in pack [Was: Number of blows not always correct]

### ISSUE THREAD

**Issue by angband-import** [angband/angband#1018]
**Reported by CunningGabe on 27 Feb 2008 20:49 UTC**
My unmodified strength is currently 18/134 and dex is 18/67. I'm currently wearing a Spear of Westernesse (+2) which boosts this to 18/154 and 18/87 and gives me 4 blows. While wearing the Spear of Westernesse, if I look at another weapon (The Morning Star 'Bllodspike') to see how many blows it will give me, the number is inflated because it is taking my modified strength and dex into account (Bloodspike gives +4 STR and no DEX bonus.) Blow-calculation should take into account the fact that I have to unwield whatever weapon I'm currently wielding.

**Comment by angband-import** [comment]
**Attachment from anonymous on 4 Nov 2008 22:08 UTC**

REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_498_GIT_1018/average_damage

**Comment by angband-import** [comment]
**Attachment from non on 21 Dec 2008 18:48 UTC**
this should only take bonus dmg into account when it's known
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_498_GIT_1018/dmginfo.1036.patch

**Comment by angband-import** [comment]
**Comment by cunninggabe on 14 Mar 2008 18:25 UTC**
[c22597c] (SVN r753) mostly fixed this. The number of blows should always be correct now, but the average damage still needs to be fixed.

**Comment by angband-import** [comment]
**Modified by cunninggabe on 17 Mar 2008 22:48 UTC**

**Comment by angband-import** [comment]
**Modified by takkaria on 28 Jun 2008 10:02 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 1 Oct 2008 07:07 UTC**
in [b09ee71] (SVN r1000).

**Comment by angband-import** [comment]
**Comment by Big Al on 21 Oct 2008 23:15 UTC**
This still seems to be broken - in shops at least. Eg. the weapon shop is selling:

 * a Lucerne Hammer (2d5) (+0, +0)
 * a Lucerne Hammer (2d5) (+3, +3)

and they both show "averaging a damage of 9 against normal creatures."  The latter should be 12 damage I believe.

**Comment by angband-import** [comment]
**Comment by CunningGabe on 23 Oct 2008 16:18 UTC**
It seems to me that the average damage isn't taking into account the bonuses of the weapon - I have a Dagger (1d4) (+6, +4) and an inherent +2 to damage, and when I examine it, it claims my average damage is 4.5 per blow. This is just the 2.5 average +2 from my own bonus. I don't get the correct figure whether I'm wielding the dagger or not.

**Comment by angband-import** [comment]
**Comment by anonymous on 4 Nov 2008 22:08 UTC**
I seem to have fixed it.

**Comment by angband-import** [comment]
**Comment by takkaria on 14 Nov 2008 22:12 UTC**
fixed in [b4f50c3] (SVN r1028).

**Comment by angband-import** [comment]
**Comment by Big Al on 12 Dec 2008 06:38 UTC**
Still broken, sorry - now it shows you the average damage as if the weapon was identified.  E.g. I find a 1d4 dagger and wield it.  It tells me I will do an average damage of 10.5.  I can figure out the +todam of the weapon in this case without identifying it: 10.5 - 2.5 (for base 1d4 damage) - 3 (STR bonus) = +5 todam.

**Comment by angband-import** [comment]
**Comment by takkaria on 24 Dec 2008 02:04 UTC**
in [d702662] (SVN r1046), thanks

**Comment by angband-import** [comment]
**Comment by Big Al on 31 Dec 2008 05:00 UTC**
In [d702662] (SVN r1046), 
```
    xtra_dam = (state.dis_to_d + o_ptr->to_d) * 10; 
```
was changed to:
```
    xtra_dam = state.dis_to_d; 
    if (object_known_p(o_ptr)) xtra_dam += o_ptr->to_d * 10;
```
Shouldn't this be:
```
    xtra_dam = state.dis_to_d * 10;
    if (object_known_p(o_ptr)) xtra_dam += o_ptr->to_d * 10;
```
A quick debug-mode char test shows that a 1d4 (+0, +0) dagger and an identified (+0, +50) ring of damage does 7.5 damage per blow, not 52.5.  The x10 multiplier seems to have been removed from the patch d_m made.

**Comment by angband-import** [comment]
**Comment by takkaria on 6 Jan 2009 21:16 UTC**
fixed in [5be6498] (SVN r1073).  That's what I get for applying patches by hand!
